public/app/listeners/executePythonAlgorithm.js

```diff
@@ -35,14 +35,13 @@ const createNewResultDataset = ({
 
 const executePythonAlgorithm = (mainWindow, db) => (
   event,
-  { datasetId, algorithmId },
+  { sourceId, algorithmId, filename },
```

Comment:
maybe worth it here and in `AddExecutionForm.js` and `/actions/executions.js` to use something like `userProvidedFilename` instead of `filename`